src/StakeVault.sol

```diff
@@ -135,13 +177,11 @@ contract StakeVault is Ownable {
     }
 
     function _stake(uint256 _amount, uint256 _seconds, address _source) internal {
+        stakeManager.stake(_amount, _seconds);
         bool success = STAKING_TOKEN.transferFrom(_source, address(this), _amount);
         if (!success) {
             revert StakeVault__StakingFailed();
         }
-
-        stakeManager.stake(_amount, _seconds);
-
```

Comment:
This change is just for consistency's sake. First update accounting, then perform actual transfer (CEI pattern)